api/repositories/service_binding_repository_test.go

```diff
@@ -1238,4 +1239,114 @@ var _ = Describe("ServiceBindingRepo", func() {
 			})
 		})
 	})
+
+	Describe("GetServiceBindingDetails", func() {
+		var (
+			serviceBindingGUID   string
+			getErr               error
+			cfServiceInstance    *korifiv1alpha1.CFServiceInstance
+			serviceBindingRecord repositories.ServiceBindingDetailsRecord
+			credentialsSecret    *corev1.Secret
+		)
+
+		BeforeEach(func() {
+			serviceInstanceGUID := prefixedGUID("instance")
+			cfServiceInstance = createServiceInstanceCR(ctx, k8sClient, serviceInstanceGUID, space.Name, "service-instance-name", serviceInstanceGUID)
+			serviceBindingGUID = prefixedGUID("binding")
+
+			credentials := map[string]any{
+				"foo": "val1",
+				"bar": "val2",
+			}
+
+			credentialBytes, err := json.Marshal(credentials)
+			Expect(err).To(BeNil())
+
+			creds := map[string][]byte{
```

Comment:
You could use [ToCredentialsSecretData](https://github.com/cloudfoundry/korifi/blob/89d5234409cc06abb6462b21cd34cf6f2435212e/tools/secrets.go#L14) to create the secret data in the correct format